games/rpg/src/lib.rs

```diff
@@ -0,0 +1,1483 @@
+                                            return health
+                                        }
+                                        priority *= dec!("1.35");
+                                        if priority > dec!(1) {
+                                            enemy_health -= damage_given * self.seed_50();
+                                            if enemy_health < dec!(0) || enemy_health == dec!(0) {
+                                                return health
+                                            }
+                                            continue;
+                                        }
+                                        else {
+                                            health -= damage_taken * self.seed_50();
+                                            enemy_health -= damage_given * self.seed_50();
+                                            if health < dec!(0) || health == dec!(0) {
+                                                return health
+                                            }
+                                            if enemy_health < dec!(0) || enemy_health == dec!(0) {
+                                                return health
+                                            }
+                                            continue;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                else {
+                    enemy_health -= damage_given * self.seed_50();
+                    if enemy_health < dec!(0) || enemy_health == dec!(0) {
+                        return health
+                    }
+                    health -= damage_taken * self.seed_50();
+                    if health < dec!(0) || health == dec!(0) {
+                        return health
+                    }
+                    continue;
+                }
+            };
+        }
+        //Levelup method
+        pub fn levelup(&mut self, nft_proof: Proof) {
+            let mut nft_data: Account = nft_proof.non_fungible().data();
+            assert!(
+                nft_proof.amount() == dec!("1"),
+                "You can only level up one character at once!"
+            );
+        
+            assert!(
+                nft_proof.resource_address() == self.character_nft,
+                "Wrong resource address!"
+            );
+            let total_exp = nft_data.exp;
+            let class = nft_data.class;
+            let hp = self.char_hp.get(&class).unwrap();
+            let atk = self.char_atk.get(&class).unwrap();
+            let mag = self.char_mag.get(&class).unwrap();
+            let def = self.char_def.get(&class).unwrap();
+            let spd = self.char_spd.get(&class).unwrap();
+            // Idk if this is good, the level scaling seems wack af lmao
+            match total_exp {
```

Comment:
It seems to me like this part can be changed to be made more general instead of needing to hardcode all of the information in.